java/src/main/native/CMakeLists.txt

```diff
@@ -277,7 +289,8 @@ include_directories("${THRUST_INCLUDE}"
                     "${JNI_INCLUDE_DIRS}"
                     "${CUDF_INCLUDE}"
                     "${RMM_INCLUDE}"
-                    "${ARROW_INCLUDE}")
+                    "${ARROW_INCLUDE}"
+                    "$<$<BOOL:${cuFile_FOUND}>:${cuFile_INCLUDE_DIRS}>")
```

Comment:
Ideally there should be separate include lists for libcudfjni.so and libcufilejni.so (or whatever we're going to call it) so we're not making it easy to create unsatisfied cufile symbols in libcudfjni.so.

If we're taking out the link for now then I question the utility of adding the includes.

Reply:
Using `include_directories` is considered a cmake "antipattern" since it's a global function (https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html). This file would need some cleanup before we can have different includes for different targets.